src/wallet/partialTxProposal.ts

```diff
@@ -48,44 +60,90 @@ class PartialTxProposal {
+   *
+   * @param {HathorWallet} wallet Wallet which will provide the tokens.
+   * @param {string?} [token='00'] UID of token that is being sent
+   *
+   * @returns {UtxoExtended[]}
+   */
+  static getWalletUtxos(
+    wallet: HathorWallet,
+    token: string = HATHOR_TOKEN_CONFIG.uid,
+  ): UtxoExtended[] {
+    const historyTransactions = wallet.getFullHistory();
+    const allUtxos = [...wallet.getAllUtxos({ token })].filter(utxo => utxo.authorities === 0);
+    const allExtendedUtxos: UtxoExtended[] = [];
+    for (const utxo of allUtxos) {
+      // Since we chose the utxos from the historyTransactions, we can be sure this exists.
+      const txout = historyTransactions[utxo.txId].outputs[utxo.index];
+      allExtendedUtxos.push({
+        ...utxo,
+        tokenData: txout.token_data,
+      });
+    }
+
+    return allExtendedUtxos;
+  }
+
   /**
    * Add inputs sending the amount of tokens specified, may add a change output.
    *
    * @param {HathorWallet} wallet Wallet which will provide the tokens.
    * @param {string} token UID of token that is being sent
    * @param {number} value Quantity of tokens being sent
    * @param {Object} [options]
+   * @param {UtxoExtended[]|null} [options.changeAddress=null] If we add change, use this address instead of getting a new one from the wallet.
    * @param {string|null} [options.changeAddress=null] If we add change, use this address instead of getting a new one from the wallet.
    * @param {boolean} [options.markAsSelected=true] Mark the utxo with `selected_as_input`.
    */
   addSend(
     wallet: HathorWallet,
     token: string,
     value: number,
-    { changeAddress = null, markAsSelected = true }: { changeAddress?: string|null, markAsSelected?: boolean } = {},
+    {
+      utxos = null,
+      changeAddress = null,
+      markAsSelected = true,
+    }: { utxos?: UtxoExtended[]|null, changeAddress?: string|null, markAsSelected?: boolean } = {},
   ) {
     this.resetSignatures();
 
-    const historyTransactions = wallet.getFullHistory();
+    // Since the selectUtxos returns a list of Utxo
+    // we need a way to access the original utxo for the tokenData.
+    const utxosDict: Record<string, UtxoExtended> = {};
+    // Use the pool of utxos or all wallet utxos.
+    const allUtxos: UtxoExtended[] = (utxos && utxos.length > 0)
+      ? utxos
+      : PartialTxProposal.getWalletUtxos(wallet, token);
+
+    // Filter pool of utxos for only utxos from the token and not already in the partial tx
+    const currentUtxos = this.partialTx.inputs.map(input => `${input.hash}-${input.index}`);
+    const utxosToUse = allUtxos.filter(utxo => utxo.tokenId === token && !currentUtxos.includes(`${utxo.txId}-${utxo.index}`));
+
+    for (const utxo of utxosToUse) {
+      utxosDict[`${utxo.txId}-${utxo.index}`] = utxo;
+    }
+    const utxosDetails = transactionUtils.selectUtxos(utxosToUse, value);
 
-    const utxosDetails = wallet.getUtxosForAmount(value, { token });
```

Comment:
Why don't you continue using this? Was it a bug previously?

Reply:
No, this method fetched and filtered utxos, but since we have a pool of utxos as an argument I thought we should call the filter method here, either for fetched utxos or argument utxos.